### PR TITLE
Codefactor duplication issues

### PR DESCRIPTION
**Pull request by SotiriaSte**
**What kind of change does this PR introduce?** 

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [X] Refactor
- [ ] Build-related changes
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [X] No

**The PR fulfills these requirements:**

- [X] It's submitted to the `main` branch
- [ ] When resolving a specific issue, it's referenced in the PR's description  (e.g. `fix #xxx`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature 

**Other information:**

### CODE REVIEW

File: modules/database/src/interfaces/ICms.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  enabled: boolean;
+  authenticated: boolean;
+}
+
+const crudOperationDefaults: CrudOperation = {
+  enabled: false,
+  authenticated: false,
+};
+
+export interface ICms {
```
Comment (kon14) [inline modules/database/src/interfaces/ICms.ts:11]: Rename this to `CmsOptions` or similar.

File: modules/database/src/interfaces/ICms.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  crudOperations: {
+    create: typeof crudOperationDefaults;
+    read: typeof crudOperationDefaults;
+    update: typeof crudOperationDefaults;
+    delete: typeof crudOperationDefaults;
+  };
+}
+
+export const cmsInstance: ICms = {
```
Comment (kon14) [inline modules/database/src/interfaces/ICms.ts:23]: Rename this to `introspectedSchemaCmsOptionsDefaults`, `defaultIntrospectedSchemaCmsOptions` or similar.

File: modules/database/src/interfaces/ICms.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+interface CrudOperation {
```
Comment (kon14) [inline modules/database/src/interfaces/ICms.ts:1]: Do we really need this extra type? You're using type inference in `ICms` anyway.

**Review by kon14 — CHANGES_REQUESTED** [review]

File: modules/database/src/interfaces/CmsOptions.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+export interface CmsOptions {
+  enabled: boolean;
+  authentication: boolean;
+  crudOperations: {
+    [key: string]: {
```
Comment (kon14) [inline modules/database/src/interfaces/CmsOptions.ts:5]: Don't drop the key signature. I only requested that the inner type be removed (if unused).
Switch back to a typed object for `crudOperations` keeping the `typeof` inference.

**Review by kon14 — CHANGES_REQUESTED** [review]

**Review by kkopanidis — APPROVED** [review]

**Pull request merged**